app/models/aws/receive_application_event.rb

```diff
@@ -0,0 +1,52 @@
+module Aws
+  class ReceiveApplicationEvent
+    attr_reader :id, :data
+
+    alias causation_id id
+
+    def initialize(id:, data:)
+      @id = id
+      @data = data
+    end
+
+    def create!
+      Reviewing::ReceiveApplication.call(
+        application_id:,
+        parent_id:,
+        work_stream:,
+        application_type:,
+        causation_id:,
+        submitted_at:,
+        reference:
+      )
+    rescue Reviewing::AlreadyReceived
+      Rails.logger.warn('Application already received')
+    end
+
+    private
+
+    def application_id
+      data['id']
+    end
+
+    def submitted_at
+      data['submitted_at'] || Time.zone.now
```

Comment:
I think you can remove the` || Time.zone.now` here as all events should have this populated now.